src/schema/objects/formats.yaml

```diff
@@ -0,0 +1,115 @@
+  pattern: '[+-]?([0-9]+([.][0-9]*)?|[.][0-9]+)([eE][+-]?[0-9]+)?'
+boolean:
+  name: Boolean
+  description: |
+    A boolean.
+    Must be either "true" or "false".
+  pattern: '(true|false)'
+# String formats
+date:
+  name: Date
+  description: |
+    A date in the form `"YYYY-MM-DD[Z]"`,
+    where [Z] is an optional, valid timezone code.
+  pattern: '[0-9]{4}-[0-9]{2}-[0-9]{2}([A-Z]{2,4})?'
+datetime:
+  name: Datetime
+  description: |
+    A datetime in the form `"YYYY-MM-DDThh:mm:ss[.000000][Z]"`,
+    where [.000000] is an optional subsecond resolution between 1 and 6 decimal points,
+    and [Z] is an optional, valid timezone code.
+  pattern: '[0-9]{4}-[0-9]{2}-[0-9]{2}T(?:2[0-3]|[01][0-9]):[0-5][0-9]:[0-5][0-9](\.[0-9]{1,6})?([A-Z]{2,4})?'
+time:
+  name: Time
+  description: |
+    A time in the form `"hh:mm:ss"`.
+  pattern: '(?:2[0-3]|[01]?[0-9]):[0-5][0-9]:[0-5][0-9]'
+unit:
+  name: A standardized unit
+  description: |
+    A unit.
+    SI units in CMIXF formatting are RECOMMENDED
+    (see [Units](/02-common-principles.html#units)).
+
+    Currently this matches any string.
+
+    TODO: Somehow reference the actual unit options in the Units appendix.
+  pattern: '.*'
+stimuli_relative:
+  name: Path relative to the stimuli folder
+  description: |
+    A path to a stimulus file, relative to a `/stimuli` folder somewhere.
+
+    The validation for this format is minimal.
+    It simply ensures that the value is a string with any characters that may appear in a valid path,
+    without starting with "/" (an absolute path) or "stimuli/"
+    (a relative path starting with the stimuli folder, rather than relative to that folder).
+  pattern: '(?!/)(?!stimuli/)[0-9a-zA-Z/\_\-\.]+'
+dataset_relative:
+  name: Path relative to the BIDS dataset folder
+  description: |
+    A path to a file, relative to the dataset folder.
+
+    The validation for this format is minimal.
+    It simply ensures that the value is a string with any characters that may appear in a valid path,
+    without starting with "/" (an absolute path).
+  pattern: '(?!/)[0-9a-zA-Z/\_\-\.]+'
+participant_relative:
+  name: Path relative to the participant folder
+  description: |
+    A path to a file, relative to the participant's folder in the dataset.
+
+    The validation for this format is minimal.
+    It simply ensures that the value is a string with any characters that may appear in a valid path,
+    without starting with "/" (an absolute path) or "sub/"
+    (a relative path starting with the participant folder, rather than relative to that folder).
+  pattern: '(?!/)(?!sub-)[0-9a-zA-Z/\_\-\.]+'
+rrid:
+  name: Research resource identifier
+  description: |
+    A [research resource identifier](https://scicrunch.org/resources).
+  pattern: 'RRID:.+_.+'
+uri:
+  name: Uniform resource indicator
+  description: |
+    A uniform resource indicator.
+  pattern: '(([^:/?#]+):)?(//([^/?#]*))?([^?#]*)(\?([^#]*))?(#(.*))?'
+bids_uri:
```

Comment:
might be worth to sort the entries to ease locating any particular one

Reply:
Good idea. I'll sort them after merging in your recommendations.